### DISCUSSION TITLE
Expose server errors during HTTP streaming

### DISCUSSION THREAD

**Discussion by tropperstyle** [withastro/roadmap#616] [Proposal]
# Summary

Expose a way in user-land to access server errors emitted during HTTP streaming

# Background & Motivation

One downside to HTTP streaming is that the response will always be 200 OK if code clears the original page front-matter. But if an error occurs downstream in an async component, [Internal server error is appended and the response abruptly ended](https://github.com/withastro/astro/blob/astro%402.8.2/packages/integrations/node/src/nodeMiddleware.ts#L60-L67).

# Goals
- At a minimum, allow to log these errors
- As an extra, allow the user to customize this behavior like providing an error overlay similar to `astro dev`

# Example

Would this be adapter-dependent? Or something for middleware?

**Comment by mb21** [comment]
I discovered this as well: https://github.com/withastro/astro/issues/8714
